src/sentry/auth/providers/saml2/provider.py

```diff
@@ -170,8 +170,6 @@ def dispatch(self, request, organization_slug):
         should_logout = request.user.is_authenticated()
 
         def force_logout():
-            request.user.refresh_session_nonce()
```

Comment:
This is being done in the signal now from normal logout, otherwise it'd rotate twice here.

Reply:
out of scope, but only concern id have is if this doesnt have a test to prove it rotates nonce and someone breaks it in the future

Reply:
In theory, just added a test for this.